src/base/bittorrent/session.cpp

```diff
@@ -155,6 +155,19 @@ namespace
 
         return expanded;
     }
+
+    QStringList findAllFiles(const QString &folderPath)
+    {
+        QStringList files;
```

Comment:
>Why not `QDirIterator`

Damn, I totally forgot about it. Fixed.

As for possible problems with access rights etc.
 I'm not going to solve it, but just give at the mercy of the libtorrent (as it was until now). Under normal conditions, we should not have such problems, and I'm not going to try to anticipate and correct all possible problems. If you really dwell on it, we get a program that will "wiping your ass", but will not do its job.